### PR TITLE
Provide friendlier error messages

### PR DESCRIPTION
**Pull request by markryall**
This attempts to provide more user friendly error messages when the attempt to store the association between a site and telegram channel/group fails.

This addresses issue #2.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -62,11 +73,12 @@ bot.on( 'message', msg => {
 		} )
 		.then( () => followBlog( msg.chat.id, 'group', url ) )
 		.then( () => bot.sendMessage( msg.chat.id, 'Following!' ) )
-		.catch( error => bot.sendMessage( msg.chat.id, 'Error: ' + error.message ) );
+		.catch( error => handleError( error, msg.chat.id, urlfix/friendly-error-messages ) );
```
Comment (yurynix) [inline index.js]: What's `urlfix/friendly-error-messages` ? should be `url`?

File: index.js [hunk 2]
```diff
@@ -43,7 +43,18 @@ function getUrlFromMsgText( msgText ) {
 	return null;
 }
 
+function handleError( error, id, url ) {
+	debug( error.message );
+
+	if ( error.message.includes( 'E11000 duplicate key' )) {
```
Comment (yurynix) [inline index.js]: Super minor, IMO, We can check `error.code === 11000` here instead

**Comment by markryall** [comment]
@yurynix thanks - that makes sense.

There was also a merge conflict I had to sort out.

File: index.js [hunk 3]
```diff
@@ -74,16 +84,17 @@ bot.on( 'message', msg => {
...
-				return Promise.reject( new Error( 'You need to be an administrator of the channel to do that' ) );
-			}
-		} )
-		.then( () => followBlog( msg.chat.id, 'group', url ) )
-		.then( () => bot.sendMessage( msg.chat.id, 'Following!' ) )
-		.catch( error => bot.sendMessage( msg.chat.id, 'Error: ' + error.message ) );
-
+			return Promise.reject( new Error( 'You need to be an administrator of the channel to do that' ) );
+		}
+		return Promise.resolve();
```
Comment (yurynix) [inline index.js]: This isn't really required

**Comment by yurynix** [comment]
Thank you for this! 😍

**Pull request merged**